plugins/blocklog_plugin/blocklog_plugin.cpp

```diff
@@ -0,0 +1,140 @@
+/**
+ *  @file
+ *  @copyright defined in eos/LICENSE.txt
+ */
+#include <eosio/blocklog_plugin/blocklog_plugin.hpp>
+#include <eosio/chain/block_log.hpp>
+#include <eosio/chain/exceptions.hpp>
+#include <eosio/chain/config.hpp>
+#include <eosio/chain/controller.hpp>
+#include <eosio/chain/generated_transaction_object.hpp>
+
+#include <eosio/chain/eosio_contract.hpp>
+
+#include <eosio/utilities/key_conversion.hpp>
+#include <eosio/utilities/common.hpp>
+
+#include <boost/algorithm/string.hpp>
+#include <boost/lexical_cast.hpp>
+
+#include <fc/io/json.hpp>
+#include <fc/variant.hpp>
+#include <signal.h>
+
+namespace eosio {
+using namespace chain;
+
+static appbase::abstract_plugin& _bnet_plugin = app().register_plugin<blocklog_plugin>();
+
+
+using namespace eosio;
+using namespace eosio::chain;
+using namespace eosio::chain::config;
+using fc::flat_map;
+
+struct blocklog_plugin_impl {
+   blocklog_plugin_impl()
+   {}
+
+   void read_log();
+
+   bfs::path                        blocks_dir;
+   bfs::path                        output_file;
+   fc::optional<block_log>          block_logger;
+   uint32_t                         first_block;
+   uint32_t                         last_block;
+};
+
+void blocklog_plugin_impl::read_log() {
+   block_logger.emplace(block_log(blocks_dir));
+   auto end = block_logger->read_head();
+   EOS_ASSERT( end, block_log_exception, "No blocks found in block log" );
+   EOS_ASSERT( end->block_num() > 1, block_log_exception, "Only one block found in block log" );
+
+   auto end_time = end->timestamp.to_time_point();
+   ilog( "existing block log with ${n} blocks", ("n",end->block_num()) );
+
+   std::ofstream output_blocks;
+   std::ostream* out;
+   if (!output_file.empty()) {
+      output_blocks.open(output_file.generic_string().c_str());
+      if (output_blocks.fail()) {
+         std::ostringstream ss;
+         ss << "Unable to open file '" << output_file.string() << "'";
+         throw std::runtime_error(ss.str());
+      }
+      out = &output_blocks;
+   }
+   else
+      out = &std::cout;
+
+   uint32_t block_num = (first_block < 1) ? 1 : first_block;
+   const uint32_t end_block = (last_block < end->block_num() ? last_block : end->block_num()) + 1;
+   signed_block_ptr next;
+   while((block_num < end_block) && (next = block_logger->read_block_by_num( block_num ))) {
+      fc::variant pretty_output;
+      abi_serializer::to_variant(*next,
+                                 pretty_output,
+                                 []( account_name n ) { return optional<abi_serializer>(); },
+                                 fc::microseconds(1000 * 1000));
```

Comment:
use fc::microseconds::maximum()

Reply:
Actually that doesn't work, so how about `fc::seconds(10)`

Reply:
```
   signed_block_ptr next;
   fc::variant pretty_output;
   const fc::microseconds deadline = fc::seconds(10);
   while((block_num < end_block) && (next = block_logger->read_block_by_num( block_num ))) {
      abi_serializer::to_variant(*next,
                                 pretty_output,
                                 []( account_name n ) { return optional<abi_serializer>(); },
                                 deadline);
      const auto block_id = next->id();
      const uint32_t ref_block_prefix = block_id._hash[1];
      auto enhanced_object = fc::mutable_variant_object
                 ("block_num",next->block_num())
                 ("id", block_id)
                 ("ref_block_prefix", ref_block_prefix)
                 (pretty_output.get_object());
      fc::variant v( std::move(enhanced_object) );
      *out << fc::json::to_pretty_string(v) << "\n";
      ++block_num;
   }
```

Reply:
Some preformance improvements. Also if you made pretty output optional you could write directly to output using `fc::to_stream( *out, v, fc::json::stringify_large_ints_and_doubles )`